src/ethereum/ethTypes.ts

```diff
@@ -496,3 +505,34 @@ export interface EthereumOtherMethods {
   ) => Promise<void>
   wcGetConnections: () => Dapp[]
 }
+
+export type SafeEthWalletInfo = ReturnType<typeof asSafeEthWalletInfo>
+export const asSafeEthWalletInfo = asSafeCommonWalletInfo
+
+export interface EthPrivateKeys {
+  mnemonic: string
+  privateKey: string
+}
+export const asEthPrivateKeys = (pluginId: string): Cleaner<EthPrivateKeys> =>
+  asCodec(
+    (value: unknown) => {
+      const from = asObject({
+        [`${pluginId}Mnemonic`]: asString,
+        [`${pluginId}Key`]: asString
+      })(value)
+      const to = {
+        mnemonic: from[`${pluginId}Mnemonic`],
+        privateKey: from[`${pluginId}Key`]
+      }
+      return asObject({
+        mnemonic: asString,
+        privateKey: asString
+      })(to)
```

Comment:
Apparently, according to the block of code that got deleted in the Ethereum engine constructor, there are keys out there that are double-nested. Something like this:

```
{
  type: "wallet:ethereum",
  id: "...",
  keys: {
    keys: {
      ethereumKey: '0x...'
    }
  }
}
```

So I think we should preserve the ability to read these jacked-up keys by fixing them in the cleaner.